strawberryfields/api/result.py

```diff
@@ -107,6 +102,43 @@ def state(self):
             raise AttributeError("The state is undefined for a stateless computation.")
         return self._state
 
+    @staticmethod
+    def combine_samples(samples_list, mode_order):
+        """Combine samples measured at different times into one nested array.
+
+        Args:
+            samples_list (list[array, tensor]): the sample measurements in a list
+            mode_order (list[array]): The nested mode order for the measured modes. Must have the
+                same structure as samples_list.
+
+        Returns:
+            array: the samples in ascending mode order with shape ``(shots, measured_modes)``
+        """
+        mode_order_flat = np.hstack(mode_order)
+        shots = len(samples_list[0])
+        modes = len(mode_order_flat)
+
+        # create the correct mode order if not all modes are measured
+        if np.max(mode_order_flat) + 1 != modes:
+            shapes = np.cumsum([len(m) for m in mode_order])
+            mode_order = np.split(
+                # argsort twice returns the rank
+                np.argsort(np.argsort(mode_order_flat)),
+                shapes,
+            )[:-1]
+
+        # find the "widest" type in samples_list
+        type_list = ["c", "f", "i"]
+        while type_list[0] not in [np.array(s).dtype.kind for s in samples_list]:
+            del type_list[0]
+
+        ret = np.empty([shots, modes], dtype=type_list[0] + "8")
```

Comment:
Nice! I wasn't aware of the string representation of NumPy dtypes, learnt something new.

Reply:
I just learnt about it as well (with a bit of trial-and-error). `dtype="c"` doesn't really work as a though, so I had to add the `"8"`, also making the `int` and `float` type into 64 bit types, while `dtype="i"` and `dtype="f"` would create 32 bit types. :slightly_smiling_face: